### ISSUE TITLE
it does not start with a readonly config storage

### ISSUE THREAD

**Issue by mc36** [asbru-cm/asbru-cm#968]
**Describe the bug**
running the appimage with a read-only mounted config dir shows the window then exits

**To Reproduce**
sudo mount -o remount,ro /your/config/path
start the app  --readonly --config-dir=/your/config/path

**Expected behavior**
it should start without any issues

**Screenshots**
mc36@ideapad:~$ ./asbru-cm-6.4.0-alpha.3.AppImage --readonly --config-dir=/nfs/own/asbru/
Gtk-Message: 17:05:13.377: Failed to load module "xapp-gtk3-module"
INFO: Ásbrú Connection Manager 6.4.0 (asbru-cm) started with PID 4590
INFO: Language detected : en_US.UTF-8, forcing to en_US.UTF-8
INFO: Desktop environment detected : cinnamon
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.9': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.8': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.7': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.6': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.5': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.4': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.3': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.2': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.1': Read-only file system
cp: cannot remove '/nfs/own/asbru//bak/asbru.yml.0': Read-only file system
INFO: Config directory is '/nfs/own/asbru/'
INFO: Used config file '/nfs/own/asbru//asbru.nfreeze'
INFO: Virtual terminal emulator (VTE) version is 0.62
INFO: Theme directory is '/tmp/.mount_asbru-p9ct0i/opt/asbru-cm/res/themes/default'
INFO: Starting '/tmp/.mount_asbru-p9ct0i/opt/asbru-cm/asbru-cm' in READ ONLY mode!
INFO: Trying to loading Unity specific tray icon package...
WARNING: AppIndicator is missing --> there might be no icon showing up in the status bar when running Unity!
INFO: Using Gnome tray icon
Use of uninitialized value $action in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 430.
Use of uninitialized value $action in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 430.
Use of uninitialized value $action in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 430.
Use of uninitialized value $action in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 430.
Use of uninitialized value in hash element at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 339.
Use of uninitialized value in hash element at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 339.
Use of uninitialized value in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 347.
Use of uninitialized value in hash element at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 339.
Use of uninitialized value in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 347.
Use of uninitialized value in hash element at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 339.
Use of uninitialized value in concatenation (.) or string at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACKeyBindings.pm line 347.
can't create /nfs/own/asbru//tmp/asbru_PID{4590}_n2freeze: Read-only file system at /tmp/.mount_asbru-p9ct0i/opt/asbru-cm/lib/PACTerminal.pm line 415.
mc36@ideapad:~$

**Environment (please complete the following information):**
 - OS: debian sid
 - Ásbrú Version 6.4.0alpha4

**Additional context**
n/a

**Comment by gfrenoy** [comment]
There are some temporary files that are created so you do need a writable path to start connection.  Did you try to specify the ``--temp-dir`` to a writable path ?

And maybe the automated backup option is causing this, we might need to disable it when `--readonly` is specified.

Let's keep this for later.

**Comment by mc36** [comment]
no, i haven't tried the --temp-dir knob... finally i moved my configs to a writeable share...

**Comment by gfrenoy** [comment]
I tested `--temp-dir` but it won't work since, until now, some temporary files are still written in a subdirectory `tmp` of the configuration file.  I'll clean up that part of the code so that we don't use any writable temporary files in the configuration directory.  Stay tuned.

**Comment by gfrenoy** [comment]
The `--readonly` option is available on the loki branch and will be available in the next release.
